contracts/strategies/dnsVectoryStrategy/DnsVectorStrategyAumLib.sol

```diff
@@ -0,0 +1,108 @@
+import "./DnsVectorStrategyStorageLib.sol";
+import "../../common/interfaces/IAum.sol";
+
+library DnsVectorStrategyAumLib {
+    function getAssetBalances() external view returns (Balance[] memory) {
+        DnsVectorStorage storage strategyStorage = DnsVectorStorageLib
+            .getStorage();
+        Balance[] memory assetBalances = new Balance[](2);
+        assetBalances[0] = Balance(
+            address(strategyStorage.aAaveSupplyToken),
+            strategyStorage.aAaveSupplyToken.balanceOf(address(this))
+        );
+        assetBalances[1] = Balance(
+            strategyStorage.vectorPoolHelperJoe.stakingToken(),
+            strategyStorage.vectorPoolHelperJoe.balanceOf(address(this))
+        );
+
+        return assetBalances;
+    }
+
+    function getLiabilityBalances() external view returns (Balance[] memory) {
+        DnsVectorStorage storage strategyStorage = DnsVectorStorageLib
+            .getStorage();
+        Balance[] memory liabilityBalances = new Balance[](1);
+        liabilityBalances[0] = Balance(
+            address(strategyStorage.vAaveBorrowToken),
+            strategyStorage.vAaveBorrowToken.balanceOf(address(this))
+        );
+        return liabilityBalances;
+    }
+
+    function getAssetValuations(bool shouldMaximise, bool shouldIncludeAmmPrice)
+        external
+        view
+        returns (Valuation[] memory)
+    {
+        DnsVectorStorage storage strategyStorage = DnsVectorStorageLib
+            .getStorage();
+        Valuation[] memory assetValuations = new Valuation[](2);
+        // assuming aaveSupplyToken == depositToken
+        assetValuations[0] = Valuation(
+            address(strategyStorage.aAaveSupplyToken),
+            strategyStorage.aAaveSupplyToken.balanceOf(address(this))
+        );
+
+        // get Traker Joe LP token reserves
+        // asuming a certain order of tokens
+        (
+            uint112 reserve0, // wAvax (generally it can be aaveSupplyToken or aaveBorrowToken)
+            uint112 reserve1, // usdc (generally it can be aaveSupplyToken or aaveBorrowToken)
+
+        ) = strategyStorage.traderJoePair.getReserves();
+        uint256 lpTokenTotalSupply = strategyStorage
+            .traderJoePair
+            .totalSupply();
+        uint256 lpTokenContractBalance = strategyStorage
+            .vectorPoolHelperJoe
+            .balanceOf(address(this));
+
+        // get the aaveBorrowToken AUM in depositToken
+        uint256 lpBorrowTokenAumInDepositToken = (uint256(reserve0) *
+            lpTokenContractBalance *
+            strategyStorage.priceOracle.getPrice(
+                strategyStorage.aaveBorrowToken,
+                shouldMaximise,
+                shouldIncludeAmmPrice
+            )) /
+            lpTokenTotalSupply /
+            10**18;
+
+        // assuming ammPairDepositToken == depositToken
+        // get the ammPairDepositToken AUM in depositToken
+        uint256 lpPairDepositAumInDepositCurrency = (uint256(reserve1) *
```

Comment:
lpPairDepositAumInDepositCurrency is the currency that you deposit into the pool when you provide liquidity along with the other currency called aaveBorrowToken

let me know if you know a better name

Reply:
Isn't this value means lpPairDepositAum in DepositToken?

Reply:
Isn't this value means lpPairDepositAum in DepositToken?

Reply:
currently our deposit currency is USDC as we all know it, and currently our lpPairDepositToken is also USDC.

However this will change in the future, so we might end up with:
lpPairDepositAumInDepositCurrency 
lpPairDepositAumInPairDepositCurrency (or just lpPairDepositAum)

maybe I am a bit too clear with the naming here, but I am trying to not force people what currency I am calculating in

Reply:
What I still don't understand is lpBorrowTokenAumInDepositToken is DepositToken while this one is DepositCurrency. What is their difference?

Reply:
ah ok, I used them as synonyms, I might clean that up soon